src/main/java/com/unstoppabledomains/exceptions/NamingServiceException.java

```diff
@@ -0,0 +1,47 @@
+package com.unstoppabledomains.exceptions;
+
+public class NamingServiceException extends Exception {
+  private static final long serialVersionUID = 1L;
+  private NSExceptionCode code;
+  private NSExceptionParams params;
+  private Throwable cause;
+
+  public NamingServiceException(NSExceptionCode code, NSExceptionParams params, Throwable cause) {
+    super(messageFromCode(code, params), cause);
+    this.code = code;
+    this.params = params;
+    this.cause = cause;
+  }
+  public NamingServiceException(NSExceptionCode code, NSExceptionParams params) {
+    super(messageFromCode(code, params));
+    this.code = code;
+    this.params = params;
+    this.cause = null;
+  }
+  public NSExceptionCode getCode() { return this.code; }
+  public String getMessage() { return messageFromCode(this.code, this.params); }
```

Comment:
We don't need this method. It is already defined for `Throwable` and should just work fine.